scheduler.py

```diff
@@ -71,3 +72,24 @@ def get_task_completion(self, task):
     def sigterm_handler(self, signum, frame):
         logging.info("Exiting gracefully.")
         sys.exit(0)
+
+    def get_worker(self):
+        if self.schedulerMode == 'Random':
+            if self.globalIncrementalWorkerId > 1:
+                random_worker_id = random.randint(0, self.globalIncrementalWorkerId - 1)
+            else:
+                random_worker_id = 0
+
+            return self.workers[random_worker_id]
+        
+        elif self.schedulerMode == 'RoundRobin':
+            if len(self.workerQueue) > 0:
+                with self.workerQueueLock:
+                    assignedWorkerId = self.workerQueue.pop(0)
+            else:
+                logging.info(f"No worker found.")
+                return None
+        
+            worker =  self.workers[assignedWorkerId]
+            self.workerQueue.append(assignedWorkerId) # Assign the worker id back into queue at the end
```

Comment:
we need to remove workers from this queue too. when a worker exits.

Reply:
As discussed offline, this would come in later iterations once we have support in driver to send us the data